# Exploring Ramen Ratings using SQL


![Soy sauce dumpling ramen at Shimomae Shouten, Japan](frank-from-5-am-ramen-fHSTIBefNXQ-unsplash.jpg)

## Step 1: Import Packages 📦
To query the data using SQL, I will require the following packages to help me!

In [1]:
import pandas as pd
import sqlite3

## Step 2: Import the Data ⬇️

In [2]:
# Importing the CSV file 
df = pd.read_csv("ramen-ratings.csv")
df.head()

Review #           Brand  \
0      2580       New Touch   
1      2579        Just Way   
2      2578          Nissin   
3      2577         Wei Lih   
4      2576  Ching's Secret   

                                             Variety Style Country Stars  \
0                          T's Restaurant Tantanmen    Cup   Japan  3.75   
1  Noodles Spicy Hot Sesame Spicy Hot Sesame Guan...  Pack  Taiwan     1   
2                      Cup Noodles Chicken Vegetable   Cup     USA  2.25   
3                      GGE Ramen Snack Tomato Flavor  Pack  Taiwan  2.75   
4                                    Singapore Curry  Pack   India  3.75   

  Top Ten  
0     NaN  
1     NaN  
2     NaN  
3     NaN  
4     NaN

## Step 3: Creating Connection Between Database and Dataset 🔗

In [3]:
# We will create a sqlite3 library and create a connection
conn = sqlite3.connect("ramen.db") # the xxxx.db can be anything

In [4]:
# After executing df.to_sql(), the contents of the DataFrame df will be written to a table named "ramen" in the SQL database represented by the connection object conn.
df.to_sql("ramen", conn) # The database name will be called "ramen" > we can name this on our own

2580

#### The `%load_ext sql` command is used to load SQL magic commands in Jupyter Notebook/IPython, allowing us to execute SQL queries directly within the notebook.

In [5]:
# Load the sql modules to iPython
%load_ext sql

# It seems like you're trying to connect to an SQLite database named ramen.db using SQL magic commands in an IPython environment. 
# This command will establish a connection to the SQLite database, allowing you to execute SQL queries directly within your IPython environment.
%sql sqlite:///ramen.db

## Step 4: Exploring the Ramen Dataset 🔍

#### Querying the Dataset using SQL:

In [6]:
%%sql

SELECT *
FROM ramen
LIMIT 5;

 * sqlite:///ramen.db
Done.


[(0, 2580, 'New Touch', "T's Restaurant Tantanmen ", 'Cup', 'Japan', '3.75', None),
 (1, 2579, 'Just Way', 'Noodles Spicy Hot Sesame Spicy Hot Sesame Guan-miao Noodles', 'Pack', 'Taiwan', '1', None),
 (2, 2578, 'Nissin', 'Cup Noodles Chicken Vegetable', 'Cup', 'USA', '2.25', None),
 (3, 2577, 'Wei Lih', 'GGE Ramen Snack Tomato Flavor', 'Pack', 'Taiwan', '2.75', None),
 (4, 2576, "Ching's Secret", 'Singapore Curry', 'Pack', 'India', '3.75', None)]

#### Question 1: How any entries are there?

In [7]:
%%sql

SELECT COUNT(*) AS Total_Entries
FROM ramen;

 * sqlite:///ramen.db
Done.


[(2580,)]

We can see that there are a total of `2580 entries` in the Ramen Dataset. Let us delve deeper to understand the data more.

#### Question 2: What are the distinct brands of Ramen in this Dataset?

In [8]:
%%sql

SELECT DISTINCT Brand
FROM ramen
ORDER BY Brand ASC;

 * sqlite:///ramen.db
Done.


[('1 To 3 Noodles',),
 ('7 Select',),
 ('7 Select/Nissin',),
 ('A-One',),
 ('A-Sha Dry Noodle',),
 ('A1',),
 ('ABC',),
 ('Acecook',),
 ('Adabi',),
 ('Ah Lai',),
 ('Ajinatori',),
 ('Amianda',),
 ('Amino',),
 ("Annie Chun's",),
 ('Aroi',),
 ('Asia Gold',),
 ('Asian Thai Foods',),
 ('Authentically Asian',),
 ('Azami',),
 ('Baijia',),
 ('Baixiang Noodles',),
 ('Baltix',),
 ('Bamee',),
 ('Batchelors',),
 ('Binh Tay',),
 ('Bon Go Jang',),
 ('Bonasia',),
 ('Boss',),
 ('CJ CheilJedang',),
 ("Campbell's",),
 ('Cap Atoom Bulan',),
 ('CarJEN',),
 ("Chaudhary's Wai Wai",),
 ('Chencun',),
 ('Chering Chang',),
 ('Chewy',),
 ('Chikara',),
 ('China Best',),
 ("Ching's Secret",),
 ('Chorip Dong',),
 ('ChoripDong',),
 ('Choumama',),
 ('Chuan Wei Wang',),
 ('Cintan',),
 ('Conimex',),
 ('Crystal Noodle',),
 ('Curry Prince',),
 ('Daddy',),
 ('Daifuku',),
 ('Daikoku',),
 ('Daraz',),
 ('Deshome',),
 ('Doll',),
 ('Dongwon',),
 ("Dr. McDougall's",),
 ('Dragonfly',),
 ('Dream Kitchen',),
 ('E-Zee',),
 ('E-mi',),
 ('Eat & Go',),
 ('Econsave',),
 ('Emart',),
 ('FMF',),
 ('Fantastic',),
 ("Farmer's Heart",),
 ('Fashion Food',),
 ('Fashion Foods',),
 ('Foodmon',),
 ('Forest Noodles',),
 ('Fortune',),
 ('Four Seas',),
 ('Fu Chang Chinese Noodle Company',),
 ('Fuji Mengyo',),
 ('Fujiwara',),
 ('Fuku',),
 ('GGE',),
 ('GS25',),
 ('GaGa',),
 ('Gau Do',),
 ('Gefen',),
 ('Global Inspiration',),
 ('Goku Uma',),
 ('Goku-Uma',),
 ('Golden Mie',),
 ('Golden Wheat',),
 ('Golden Wonder',),
 ('Gomex',),
 ('Good Tto Leu Foods',),
 ('Great Value',),
 ('GreeNoodle',),
 ('Guava Story',),
 ('Haioreum',),
 ("Han's South Korea",),
 ('Hankow',),
 ('Hao Way',),
 ('Happy Cook',),
 ('Happy Family',),
 ('Healtimie',),
 ('Hi-Myon',),
 ('Higashi',),
 ('Higashimaru',),
 ('HoMyeonDang',),
 ('Hosoonyi',),
 ('Hsin Tung Yang',),
 ('Hua Feng',),
 ('Hua Feng Noodle Expert',),
 ('ICA',),
 ('IbuRamen',),
 ('Ibumie',),
 ('Ikeda Shoku',),
 ('Indomie',),
 ('Ishimaru',),
 ('Itomen',),
 ('Itsuki',),
 ('J.J.',),
 ('JFC',),
 ('JML',),
 ('Jackpot Teriyaki',),
 ('Jingqi',),
 ('Just Way',),
 ('KOKA',),
 ('Kabuto Noodles',),
 ('Kailo',),
 ('Kamfen',),
 ('Kang Shi Fu',),
 ('Katoz',),
 ('Kiki Noodle',),
 ('Kim Ve Wong',),
 ("Kim's Bowl",),
 ('Kimura',),
 ('Kin-Dee',),
 ('Knorr',),
 ('Ko-Lee',),
 ('Koh Thai',),
 ('Koka',),
 ('Komforte Chockolates',),
 ('Koyo',),
 ('Kumamoto',),
 ('Kuriki',),
 ('La Fonte',),
 ('La Moderna',),
 ('Lee Fah Mee',),
 ('Lele',),
 ('Liang Cheng Mai',),
 ('Lipton',),
 ('Lishan',),
 ('Lishan Food Manufacturing',),
 ('Little Cook',),
 ('Liu Quan',),
 ('Long Jun Hang',),
 ('Long Kow',),
 ('Lotus Foods',),
 ('Love Cook',),
 ('Lucky Me!',),
 ('MAMA',),
 ('Maggi',),
 ('Maitri',),
 ('Mama',),
 ("Mama Pat's",),
 ('Mamee',),
 ('Maruchan',),
 ('Marutai',),
 ('Master Kong',),
 ('Mee Jang',),
 ('Men-Sunaoshi',),
 ('Menraku',),
 ('Mexi-Ramen',),
 ('Mi E-Zee',),
 ('Mi Sedaap',),
 ('Mie Sedaap',),
 ('Migawon',),
 ('Miliket',),
 ('Miracle Noodle',),
 ('Mitoku',),
 ("Mom's Dry Noodle",),
 ('Morre',),
 ("Mr. Lee's Noodles",),
 ('Mr. Noodles',),
 ('Mr. Udon',),
 ('Mug Shot',),
 ('Mum Ngon',),
 ('MyKuali',),
 ('MyOri',),
 ('Myojo',),
 ('Nagao Noodle',),
 ('Nagatanien',),
 ('Nakaya Shouten',),
 ('Nan Hsing',),
 ('Nan Jie Cun',),
 ('Nanyang Chef',),
 ('New Touch',),
 ('New Way',),
 ('Nissin',),
 ('No Name',),
 ('Noah Foods',),
 ('Nongshim',),
 ('Noodle Time',),
 ('Nyor Nyar',),
 ('O Sung',),
 ('ORee Garden',),
 ('Ogasawara',),
 ('Ohsung',),
 ('Omachi',),
 ('One Dish Asia',),
 ('Oni Hot Pot',),
 ('Osaka Ramen',),
 ('Ottogi',),
 ('Oyatsu',),
 ('Paldo',),
 ('Paldo Vina',),
 ('Pama',),
 ('Pamana',),
 ('Papa',),
 ('Patanjali',),
 ('Payless',),
 ('Peyang',),
 ('Pirkka',),
 ('Plats Du Chef',),
 ('Pop Bihun',),
 ('Pot Noodle',),
 ('Pran',),
 ('Premiere',),
 ('President',),
 ('President Rice',),
 ('Prima',),
 ('Prima Taste',),
 ('Pringles',),
 ('Pulmuone',),
 ('Q',),
 ('Qin Zong',),
 ('Quickchow',),
 ('Rhee Bros Assi',),
 ('Right Foods',),
 ("Ripe'n'Dry",),
 ('Rocket Brand',),
 ('Roland',),
 ('Royal Umbrella',),

Do you see your favourite Ramen Brand in this list? Because I do!!! I'm a huge fan of Nissin and Indomee 🍜

Hmmmm, there are so many brands here. I would now be interested to know exactly how many brands are represented in the dataset.

#### Question 3: How many different brands of Ramen are there in the dataset?


In [9]:
%%sql

SELECT COUNT(*) AS num_distinct_brands
FROM (
    SELECT DISTINCT Brand
    FROM ramen
) AS distinct_brands;

 * sqlite:///ramen.db
Done.


[(355,)]

A total of `355 Ramen Brands` are represented in this dataset.

#### Question 4: What are the unique countries that are represented in this Dataset? How many are there?

In [10]:
%%sql

SELECT DISTINCT country
FROM ramen
ORDER BY country ASC;

 * sqlite:///ramen.db
Done.


[('Australia',),
 ('Bangladesh',),
 ('Brazil',),
 ('Cambodia',),
 ('Canada',),
 ('China',),
 ('Colombia',),
 ('Dubai',),
 ('Estonia',),
 ('Fiji',),
 ('Finland',),
 ('Germany',),
 ('Ghana',),
 ('Holland',),
 ('Hong Kong',),
 ('Hungary',),
 ('India',),
 ('Indonesia',),
 ('Japan',),
 ('Malaysia',),
 ('Mexico',),
 ('Myanmar',),
 ('Nepal',),
 ('Netherlands',),
 ('Nigeria',),
 ('Pakistan',),
 ('Philippines',),
 ('Poland',),
 ('Sarawak',),
 ('Singapore',),
 ('South Korea',),
 ('Sweden',),
 ('Taiwan',),
 ('Thailand',),
 ('UK',),
 ('USA',),
 ('United States',),
 ('Vietnam',)]

SINGAPORE REPRESENT!!! 🇸🇬

In [11]:
%%sql
SELECT COUNT(*) as Num_Of_Unique_Countries
FROM (SELECT 
      DISTINCT Country
      FROM ramen)
AS distinct_country;

 * sqlite:///ramen.db
Done.


[(38,)]

A total of `38 countries` are represented in this dataset

#### Question 5: What are the top-rated Ramen Brands and their average ratings?

In [12]:
%%sql

SELECT
    Brand
    , ROUND(AVG(Stars), 2) as average_rating
FROM ramen
GROUP BY Brand
ORDER BY average_rating DESC;

 * sqlite:///ramen.db
Done.


[('Torishi', 5.0),
 ('The Ramen Rater Select', 5.0),
 ('The Bridge', 5.0),
 ('Tao Kae Noi', 5.0),
 ('Takamori', 5.0),
 ('Song Hak', 5.0),
 ('Seven & I', 5.0),
 ('Prima Taste', 5.0),
 ('Prima', 5.0),
 ('Plats Du Chef', 5.0),
 ('Peyang', 5.0),
 ('Patanjali', 5.0),
 ('ORee Garden', 5.0),
 ('Nyor Nyar', 5.0),
 ('MyOri', 5.0),
 ('Komforte Chockolates', 5.0),
 ('Kimura', 5.0),
 ('Kiki Noodle', 5.0),
 ('Jackpot Teriyaki', 5.0),
 ('Higashi', 5.0),
 ('Foodmon', 5.0),
 ('Daifuku', 5.0),
 ('Daddy', 5.0),
 ('ChoripDong', 5.0),
 ('MyKuali', 4.95),
 ('CarJEN', 4.93),
 ('TTL', 4.92),
 ('Nanyang Chef', 4.88),
 ("Mom's Dry Noodle", 4.83),
 ('Tseng Noodles', 4.82),
 ("World O' Noodle", 4.75),
 ('Ogasawara', 4.75),
 ('Ah Lai', 4.75),
 ('Yamamori', 4.63),
 ('Salam Mie', 4.63),
 ('Yamachan', 4.57),
 ('The Kitchen Food', 4.5),
 ('Seven-Eleven', 4.5),
 ('Sanyo Foods', 4.5),
 ('President', 4.5),
 ('Ishimaru', 4.5),
 ('Higashimaru', 4.5),
 ('Dongwon', 4.5),
 ('Chikara', 4.5),
 ('Aroi', 4.5),
 ('Sun Noodle', 4.46),
 ('Tablemark', 4.42),
 ('Kuriki', 4.42),
 ('Happy Cook', 4.42),
 ('Jingqi', 4.38),
 ('Chering Chang', 4.35),
 ('E-Zee', 4.33),
 ('New Touch', 4.28),
 ('Xiuhe', 4.25),
 ('Sempio', 4.25),
 ('Nagao Noodle', 4.25),
 ('Menraku', 4.25),
 ('Hua Feng', 4.25),
 ('Golden Wonder', 4.25),
 ('GS25', 4.25),
 ('Chorip Dong', 4.25),
 ('SuperMi', 4.22),
 ("Vit's", 4.21),
 ('Marutai', 4.21),
 ('KOKA', 4.21),
 ('ABC', 4.21),
 ('Love Cook', 4.2),
 ('Eat & Go', 4.2),
 ('Yamamoto', 4.19),
 ('Forest Noodles', 4.19),
 ('Sakurai Foods', 4.18),
 ('Takamori Kosan', 4.16),
 ('Mi Sedaap', 4.15),
 ('Wugudaochang', 4.13),
 ('Saji', 4.13),
 ('JFC', 4.13),
 ('Ikeda Shoku', 4.13),
 ('FMF', 4.13),
 ('Deshome', 4.13),
 ('Xiao Ban Mian', 4.1),
 ('HoMyeonDang', 4.1),
 ('Indomie', 4.07),
 ('A-Sha Dry Noodle', 4.07),
 ('Dragonfly', 4.06),
 ('Kang Shi Fu', 4.05),
 ('Paldo', 4.02),
 ('Souper', 4.0),
 ('Sichuan Baijia', 4.0),
 ('Sanpo', 4.0),
 ('Samurai Ramen', 4.0),
 ('S&S', 4.0),
 ('Pran', 4.0),
 ('Nongshim', 4.0),
 ('New Way', 4.0),
 ('Nakaya Shouten', 4.0),
 ('Morre', 4.0),
 ('Miracle Noodle', 4.0),
 ('Maitri', 4.0),
 ('Liu Quan', 4.0),
 ('Lee Fah Mee', 4.0),
 ('La Fonte', 4.0),
 ('Hi-Myon', 4.0),
 ('Happy Family', 4.0),
 ('Good Tto Leu Foods', 4.0),
 ('Fu Chang Chinese Noodle Company', 4.0),
 ('Curry Prince', 4.0),
 ('Choumama', 4.0),
 ('1 To 3 Noodles', 4.0),
 ('Samyang Foods', 3.99),
 ('Uni-President', 3.96),
 ('Fujiwara', 3.96),
 ('Oni Hot Pot', 3.95),
 ('Amianda', 3.95),
 ('Super Bihun', 3.94),
 ('Mamee', 3.94),
 ('Dream Kitchen', 3.94),
 ('Nissin', 3.92),
 ('Sakura Noodle', 3.9),
 ('MAMA', 3.9),
 ('JML', 3.9),
 ('Itomen', 3.9),
 ('Sarimi', 3.89),
 ('Unif-100', 3.88),
 ('Tasty Bite', 3.88),
 ('Sugakiya Foods', 3.88),
 ('Premiere', 3.88),
 ('J.J.', 3.88),
 ('Fuku', 3.88),
 ('Daikoku', 3.88),
 ('Master Kong', 3.85),
 ('Kabuto Noodles', 3.85),
 ('Hao Way', 3.84),
 ('Sapporo Ichiban', 3.83),
 ('Ruski', 3.83),
 ('Payless', 3.83),
 ('Emart', 3.82),
 ('Tokushima Seifun', 3.81),
 ('Pulmuone', 3.81),
 ("Ching's Secret", 3.81),
 ('Adabi', 3.81),
 ('Myojo', 3.8),
 ('GaGa', 3.79),
 ('Wu-Mu', 3.77),
 ('Zow Zow', 3.75),
 ('Yum Yum', 3.75),
 ('Uncle Sun', 3.75),
 ('Tropicana Slim', 3.75),
 ('TRDP', 3.75),
 ('Sura', 3.75),
 ('Star Anise Foods', 3.75),
 ('Sanrio', 3.75),
 ('Pop Bihun', 3.75),
 ('Pama', 3.75),
 ('Osaka Ramen', 3.75),
 ('Omachi', 3.75),
 ('Miliket', 3.75),
 ('Men-Sunaoshi', 3.75),
 ('Koka', 3.75),
 ('Kin-Dee', 3.75),
 ('Ibumie', 3.75),
 ('Hosoonyi', 3.75),
 ('Healtimie', 3.75),
 ('Goku Uma', 3.75),
 ("Farmer's Heart", 3.75),
 ('E-mi', 3.75),
 ('7 Select', 3.75),
 ('Baixiang Noodles', 3.7),
 ('Thai Chef', 3.69),
 ("Mama Pat's", 3.69),
 ('Maggi', 3.68),
 ('Wu Mu', 3.64),
 ('Wei Chuan', 3.63),
 ('Oyatsu', 3.63),
 ('Mama', 3.63),
 ('Itsuki', 3.63),
 ('Hankow', 3.63),
 ('Sau Tao', 3.61),
 ('Super', 3.6),
 ('Wang', 3.58),
 ('Pirkka', 3.58),
 ('Golden Mie', 3.58),
 ('Unif', 3.55),
 ("Mr. Lee's Noodles", 3.55),
 ('Maruchan', 3.55),
 ('Asian Thai Foods', 3.55),
 ('Vina Acecook', 3.54),

Is your favourite Ramen Brand amongst the top few? 🤔

I want to find my two favourite brands, which are Nissin and Indomee. 

In [13]:
%%sql

SELECT
    Brand
    , ROUND(AVG(Stars), 2) as average_rating
FROM ramen
WHERE Brand LIKE "%Nissin%" 
    OR Brand LIKE "%Indo%"
GROUP BY Brand
ORDER BY average_rating DESC;

 * sqlite:///ramen.db
Done.


[('Indomie', 4.07), ('Nissin', 3.92), ('7 Select/Nissin', 3.5)]

Seems like the average rating for Indomie and Nissin are `relatively decent`! It is definitely good to know that they're not receiving poor ratings.

Ignoreing the 7 Select/Nissin as it is a collaboration between 7-11 and Nisson. NOT COUNTEDDD!!

#### Question 6: What are the most common style of Ramen?

In [14]:
%%sql

SELECT 
    Style
    , COUNT(*) AS Style_Count
FROM ramen
GROUP BY Style
ORDER BY Style_Count DESC;

 * sqlite:///ramen.db
Done.


[('Pack', 1531),
 ('Bowl', 481),
 ('Cup', 450),
 ('Tray', 108),
 ('Box', 6),
 (None, 2),
 ('Can', 1),
 ('Bar', 1)]

The numbers represent the quantities of instant noodle packaging sold or produced, categorized into four types: Pack, Bowl, Cup, and Tray.

1) Pack: Refers to instant noodles packaged in a traditional packet or bag format.
2) Bowl: Refers to instant noodles served in a bowl-shaped container.
3) Cup: Refers to instant noodles packaged in a cup-shaped container, often with a lid.
4) Tray: Refers to instant noodles packaged in a tray or container, typically with a plastic or foam base.

#### Question 7: Which Countries produce the highest-rated ramen? Show only the top 15 Countries.

In [15]:
%%sql

SELECT
    Country
    , ROUND(AVG(Stars),2) as Avg_Stars
FROM ramen
GROUP BY Country
ORDER BY Avg_Stars DESC
LIMIT 15;

 * sqlite:///ramen.db
Done.


[('Brazil', 4.35),
 ('Sarawak', 4.33),
 ('Cambodia', 4.2),
 ('Singapore', 4.13),
 ('Malaysia', 4.13),
 ('Indonesia', 4.07),
 ('Japan', 3.98),
 ('Myanmar', 3.95),
 ('Fiji', 3.88),
 ('Hong Kong', 3.8),
 ('South Korea', 3.77),
 ('United States', 3.75),
 ('Mexico', 3.73),
 ('Bangladesh', 3.71),
 ('Taiwan', 3.67)]

The data suggests that instant noodles from `Brazil and Sarawak have the highest average ratings`, followed closely by Cambodia, Singapore, and Malaysia. On the other hand, instant noodles from Taiwan, Bangladesh, and Mexico have comparatively lower average ratings.

This piece of information can be useful for Ramen Ethusiasts who want to try the noodles from different countries.

#### Question 8: What are the top ten recognized brands of instant noodles?

In [16]:
%%sql

ALTER TABLE ramen
RENAME COLUMN "Top Ten" TO "Top_Ten";

 * sqlite:///ramen.db
Done.


[]

In [17]:
%%sql

SELECT *
FROM ramen
WHERE Top_Ten != "None";


 * sqlite:///ramen.db
Done.


[(616, 1964, 'MAMA', 'Instant Noodles Coconut Milk Flavour', 'Pack', 'Myanmar', '5', '2016 #10'),
 (633, 1947, 'Prima Taste', 'Singapore Laksa Wholegrain La Mian', 'Pack', 'Singapore', '5', '2016 #1'),
 (655, 1925, 'Prima', "Juzz's Mee Creamy Chicken Flavour", 'Pack', 'Singapore', '5', '2016 #8'),
 (673, 1907, 'Prima Taste', 'Singapore Curry Wholegrain La Mian', 'Pack', 'Singapore', '5', '2016 #5'),
 (752, 1828, 'Tseng Noodles', 'Scallion With Sichuan Pepper  Flavor', 'Pack', 'Taiwan', '5', '2016 #9'),
 (891, 1689, 'Wugudaochang', 'Tomato Beef Brisket Flavor Purple Potato Noodle', 'Pack', 'China', '5', '2016 #7'),
 (942, 1638, 'A-Sha Dry Noodle', 'Veggie Noodle Tomato Noodle With Vine Ripened Tomato Sauce', 'Pack', 'Taiwan', '5', '2015 #10'),
 (963, 1617, 'MyKuali', 'Penang Hokkien Prawn Noodle (New Improved Taste)', 'Pack', 'Malaysia', '5', '2015 #7'),
 (995, 1585, 'CarJEN', 'Nyonya Curry Laksa', 'Pack', 'Malaysia', '5', '2015 #4'),
 (1059, 1521, 'Maruchan', 'Gotsumori Sauce Yakisoba', 'Tray', 'Japan', '5', '2015 #9'),
 (1072, 1508, 'Mamee', 'Chef Gold Recipe Mi Kari Seribu Rasa', 'Pack', 'Malaysia', '5', '2015 #6'),
 (1080, 1500, 'MyKuali', 'Penang Red Tom Yum Goong Noodle', 'Pack', 'Malaysia', '5', '2015 #1'),
 (1109, 1471, 'Mama', 'Instant Noodles Shrimp Creamy Tom Yum Flavour Jumbo Pack', 'Pack', 'Thailand', '5', '2013 #10'),
 (1114, 1466, 'Mama', 'Oriental Style Instant Noodles Green Curry Flavour Jumbo Pack', 'Pack', 'Thailand', '5', '2015 #8'),
 (1230, 1350, 'Mamee', 'Chef Curry Laksa Flavour', 'Pack', 'Malaysia', '5', '2014 #7'),
 (1250, 1330, 'Sapporo Ichiban', 'Otafuku Okonomi Sauce Yakisoba', 'Tray', 'Japan', '5', '2014 #4'),
 (1272, 1308, 'Nongshim', 'Soon Veggie Noodle Soup', 'Pack', 'South Korea', '5', '2014 #9'),
 (1278, 1302, 'Mama', 'Instant Noodles Yentafo Tom Yum Mohfai Flavour', 'Pack', 'Thailand', '5', '2014 #10'),
 (1309, 1271, 'Prima Taste', 'Singapore Chilli Crab La Mian', 'Pack', 'Singapore', '5', '2014 #8'),
 (1382, 1198, 'Samyang Foods', 'Maesaengyitangmyun Baked Noodle', 'Pack', 'South Korea', '5', '2014 #5'),
 (1397, 1183, 'Paldo', 'Cheese Noodle', 'Pack', 'South Korea', '5', '2014 #6'),
 (1493, 1087, 'MyKuali', 'Penang White Curry Noodle', 'Pack', 'Malaysia', '5', '2014 #1'),
 (1588, 992, 'Prima Taste', 'Singapore Laksa La Mian', 'Pack', 'Singapore', '5', '2013 #1'),
 (1589, 991, 'Prima Taste', 'Singapore Curry La Mian', 'Pack', 'Singapore', '5', '2013 #2'),
 (1646, 934, 'Nongshim', "Jinjja Jinjja Flamin' Hot & Nutty", 'Pack', 'USA', '5', '2013 #4'),
 (1670, 910, 'Paldo', 'Namja Ramyun', 'Bowl', 'South Korea', '4.25', '\n'),
 (1757, 823, 'Paldo', 'Kokomen Spicy Chicken', 'Pack', 'South Korea', '5', '2013 #9'),
 (1846, 734, 'Indomie', 'Mi Goreng Rasa Ayam Panggang Jumbo (Local)', 'Pack', 'Indonesia', '5', '\n'),
 (1865, 715, 'Indomie', 'Mi Goreng Rendang (Import)', 'Pack', 'Indonesia', '5', '2013 #3'),
 (1972, 608, 'Koka', 'Spicy Black Pepper', 'Pack', 'Singapore', '5', '2012 #10'),
 (2002, 578, 'Nongshim', 'Shin Ramyun Black', 'Pack', 'South Korea', '4.75', '2012 #7'),
 (2146, 434, 'Mi Sedaap', 'Kari Spesial', 'Pack', 'Indonesia', '4.5', '2012 #5'),
 (2188, 392, 'Nissin', 'Yakisoba Noodles Karashi', 'Tray', 'Japan', '5', '2012 #3'),
 (2189, 391, 'Myojo', 'Hyoubanya No Chukasoba Oriental', 'Pack', 'Japan', '4.25', '2012 #6'),
 (2248, 332, 'Nongshim', 'Champong Oriental Noodles', 'Cup', 'South Korea', '3.75', '\n'),
 (2295, 285, 'Doll', 'Artificial Chicken', 'Pack', 'Hong Kong', '4.5', '2012 #9'),
 (2475, 105, 'Indomie', 'Special Fried Curly Noodle', 'Pack', 'Indonesia', '5', '2012 #1'),
 (2533, 47, 'Indomie', 'Mi Goreng Jumbo Barbecue Chicken', 'Pack', 'Indonesia', '5', '2012 #2'),
 (2535, 45, 'Indomie', 'Mi Goreng Sate', 'Pack', 'Indonesia', '5', '\n'),
 (2549, 31, 'Myojo', 'Ippeichan Yakisoba', 'Tray', 'Japan', '4', '2013 #6'),
 (2567, 13, 'Sapporo Ichiban', 'Chow Mein', 'Pack', 'Japan', '5', '2012 #4')]

I still see those emmpty cells and it's getting annoying.... What else can I do?

In [18]:
%%sql

SELECT *
FROM ramen
WHERE LENGTH(Top_Ten) > 1
ORDER BY Top_Ten DESC;

 * sqlite:///ramen.db
Done.


[(752, 1828, 'Tseng Noodles', 'Scallion With Sichuan Pepper  Flavor', 'Pack', 'Taiwan', '5', '2016 #9'),
 (655, 1925, 'Prima', "Juzz's Mee Creamy Chicken Flavour", 'Pack', 'Singapore', '5', '2016 #8'),
 (891, 1689, 'Wugudaochang', 'Tomato Beef Brisket Flavor Purple Potato Noodle', 'Pack', 'China', '5', '2016 #7'),
 (673, 1907, 'Prima Taste', 'Singapore Curry Wholegrain La Mian', 'Pack', 'Singapore', '5', '2016 #5'),
 (616, 1964, 'MAMA', 'Instant Noodles Coconut Milk Flavour', 'Pack', 'Myanmar', '5', '2016 #10'),
 (633, 1947, 'Prima Taste', 'Singapore Laksa Wholegrain La Mian', 'Pack', 'Singapore', '5', '2016 #1'),
 (1059, 1521, 'Maruchan', 'Gotsumori Sauce Yakisoba', 'Tray', 'Japan', '5', '2015 #9'),
 (1114, 1466, 'Mama', 'Oriental Style Instant Noodles Green Curry Flavour Jumbo Pack', 'Pack', 'Thailand', '5', '2015 #8'),
 (963, 1617, 'MyKuali', 'Penang Hokkien Prawn Noodle (New Improved Taste)', 'Pack', 'Malaysia', '5', '2015 #7'),
 (1072, 1508, 'Mamee', 'Chef Gold Recipe Mi Kari Seribu Rasa', 'Pack', 'Malaysia', '5', '2015 #6'),
 (995, 1585, 'CarJEN', 'Nyonya Curry Laksa', 'Pack', 'Malaysia', '5', '2015 #4'),
 (942, 1638, 'A-Sha Dry Noodle', 'Veggie Noodle Tomato Noodle With Vine Ripened Tomato Sauce', 'Pack', 'Taiwan', '5', '2015 #10'),
 (1080, 1500, 'MyKuali', 'Penang Red Tom Yum Goong Noodle', 'Pack', 'Malaysia', '5', '2015 #1'),
 (1272, 1308, 'Nongshim', 'Soon Veggie Noodle Soup', 'Pack', 'South Korea', '5', '2014 #9'),
 (1309, 1271, 'Prima Taste', 'Singapore Chilli Crab La Mian', 'Pack', 'Singapore', '5', '2014 #8'),
 (1230, 1350, 'Mamee', 'Chef Curry Laksa Flavour', 'Pack', 'Malaysia', '5', '2014 #7'),
 (1397, 1183, 'Paldo', 'Cheese Noodle', 'Pack', 'South Korea', '5', '2014 #6'),
 (1382, 1198, 'Samyang Foods', 'Maesaengyitangmyun Baked Noodle', 'Pack', 'South Korea', '5', '2014 #5'),
 (1250, 1330, 'Sapporo Ichiban', 'Otafuku Okonomi Sauce Yakisoba', 'Tray', 'Japan', '5', '2014 #4'),
 (1278, 1302, 'Mama', 'Instant Noodles Yentafo Tom Yum Mohfai Flavour', 'Pack', 'Thailand', '5', '2014 #10'),
 (1493, 1087, 'MyKuali', 'Penang White Curry Noodle', 'Pack', 'Malaysia', '5', '2014 #1'),
 (1757, 823, 'Paldo', 'Kokomen Spicy Chicken', 'Pack', 'South Korea', '5', '2013 #9'),
 (2549, 31, 'Myojo', 'Ippeichan Yakisoba', 'Tray', 'Japan', '4', '2013 #6'),
 (1646, 934, 'Nongshim', "Jinjja Jinjja Flamin' Hot & Nutty", 'Pack', 'USA', '5', '2013 #4'),
 (1865, 715, 'Indomie', 'Mi Goreng Rendang (Import)', 'Pack', 'Indonesia', '5', '2013 #3'),
 (1589, 991, 'Prima Taste', 'Singapore Curry La Mian', 'Pack', 'Singapore', '5', '2013 #2'),
 (1109, 1471, 'Mama', 'Instant Noodles Shrimp Creamy Tom Yum Flavour Jumbo Pack', 'Pack', 'Thailand', '5', '2013 #10'),
 (1588, 992, 'Prima Taste', 'Singapore Laksa La Mian', 'Pack', 'Singapore', '5', '2013 #1'),
 (2295, 285, 'Doll', 'Artificial Chicken', 'Pack', 'Hong Kong', '4.5', '2012 #9'),
 (2002, 578, 'Nongshim', 'Shin Ramyun Black', 'Pack', 'South Korea', '4.75', '2012 #7'),
 (2189, 391, 'Myojo', 'Hyoubanya No Chukasoba Oriental', 'Pack', 'Japan', '4.25', '2012 #6'),
 (2146, 434, 'Mi Sedaap', 'Kari Spesial', 'Pack', 'Indonesia', '4.5', '2012 #5'),
 (2567, 13, 'Sapporo Ichiban', 'Chow Mein', 'Pack', 'Japan', '5', '2012 #4'),
 (2188, 392, 'Nissin', 'Yakisoba Noodles Karashi', 'Tray', 'Japan', '5', '2012 #3'),
 (2533, 47, 'Indomie', 'Mi Goreng Jumbo Barbecue Chicken', 'Pack', 'Indonesia', '5', '2012 #2'),
 (1972, 608, 'Koka', 'Spicy Black Pepper', 'Pack', 'Singapore', '5', '2012 #10'),
 (2475, 105, 'Indomie', 'Special Fried Curly Noodle', 'Pack', 'Indonesia', '5', '2012 #1')]

Yay! Problem Solved! 👍

In [19]:
%%sql

SELECT 
    *,
    SUBSTR(Top_Ten, 1, INSTR(Top_Ten, ' ') - 1) AS Year,
    SUBSTR(Top_Ten, INSTR(Top_Ten, ' ') + 1) AS Ranking
FROM ramen
WHERE LENGTH(Top_Ten) > 1 
ORDER BY Year DESC, Ranking DESC;

 * sqlite:///ramen.db
Done.


[(752, 1828, 'Tseng Noodles', 'Scallion With Sichuan Pepper  Flavor', 'Pack', 'Taiwan', '5', '2016 #9', '2016', '#9'),
 (655, 1925, 'Prima', "Juzz's Mee Creamy Chicken Flavour", 'Pack', 'Singapore', '5', '2016 #8', '2016', '#8'),
 (891, 1689, 'Wugudaochang', 'Tomato Beef Brisket Flavor Purple Potato Noodle', 'Pack', 'China', '5', '2016 #7', '2016', '#7'),
 (673, 1907, 'Prima Taste', 'Singapore Curry Wholegrain La Mian', 'Pack', 'Singapore', '5', '2016 #5', '2016', '#5'),
 (616, 1964, 'MAMA', 'Instant Noodles Coconut Milk Flavour', 'Pack', 'Myanmar', '5', '2016 #10', '2016', '#10'),
 (633, 1947, 'Prima Taste', 'Singapore Laksa Wholegrain La Mian', 'Pack', 'Singapore', '5', '2016 #1', '2016', '#1'),
 (1059, 1521, 'Maruchan', 'Gotsumori Sauce Yakisoba', 'Tray', 'Japan', '5', '2015 #9', '2015', '#9'),
 (1114, 1466, 'Mama', 'Oriental Style Instant Noodles Green Curry Flavour Jumbo Pack', 'Pack', 'Thailand', '5', '2015 #8', '2015', '#8'),
 (963, 1617, 'MyKuali', 'Penang Hokkien Prawn Noodle (New Improved Taste)', 'Pack', 'Malaysia', '5', '2015 #7', '2015', '#7'),
 (1072, 1508, 'Mamee', 'Chef Gold Recipe Mi Kari Seribu Rasa', 'Pack', 'Malaysia', '5', '2015 #6', '2015', '#6'),
 (995, 1585, 'CarJEN', 'Nyonya Curry Laksa', 'Pack', 'Malaysia', '5', '2015 #4', '2015', '#4'),
 (942, 1638, 'A-Sha Dry Noodle', 'Veggie Noodle Tomato Noodle With Vine Ripened Tomato Sauce', 'Pack', 'Taiwan', '5', '2015 #10', '2015', '#10'),
 (1080, 1500, 'MyKuali', 'Penang Red Tom Yum Goong Noodle', 'Pack', 'Malaysia', '5', '2015 #1', '2015', '#1'),
 (1272, 1308, 'Nongshim', 'Soon Veggie Noodle Soup', 'Pack', 'South Korea', '5', '2014 #9', '2014', '#9'),
 (1309, 1271, 'Prima Taste', 'Singapore Chilli Crab La Mian', 'Pack', 'Singapore', '5', '2014 #8', '2014', '#8'),
 (1230, 1350, 'Mamee', 'Chef Curry Laksa Flavour', 'Pack', 'Malaysia', '5', '2014 #7', '2014', '#7'),
 (1397, 1183, 'Paldo', 'Cheese Noodle', 'Pack', 'South Korea', '5', '2014 #6', '2014', '#6'),
 (1382, 1198, 'Samyang Foods', 'Maesaengyitangmyun Baked Noodle', 'Pack', 'South Korea', '5', '2014 #5', '2014', '#5'),
 (1250, 1330, 'Sapporo Ichiban', 'Otafuku Okonomi Sauce Yakisoba', 'Tray', 'Japan', '5', '2014 #4', '2014', '#4'),
 (1278, 1302, 'Mama', 'Instant Noodles Yentafo Tom Yum Mohfai Flavour', 'Pack', 'Thailand', '5', '2014 #10', '2014', '#10'),
 (1493, 1087, 'MyKuali', 'Penang White Curry Noodle', 'Pack', 'Malaysia', '5', '2014 #1', '2014', '#1'),
 (1757, 823, 'Paldo', 'Kokomen Spicy Chicken', 'Pack', 'South Korea', '5', '2013 #9', '2013', '#9'),
 (2549, 31, 'Myojo', 'Ippeichan Yakisoba', 'Tray', 'Japan', '4', '2013 #6', '2013', '#6'),
 (1646, 934, 'Nongshim', "Jinjja Jinjja Flamin' Hot & Nutty", 'Pack', 'USA', '5', '2013 #4', '2013', '#4'),
 (1865, 715, 'Indomie', 'Mi Goreng Rendang (Import)', 'Pack', 'Indonesia', '5', '2013 #3', '2013', '#3'),
 (1589, 991, 'Prima Taste', 'Singapore Curry La Mian', 'Pack', 'Singapore', '5', '2013 #2', '2013', '#2'),
 (1109, 1471, 'Mama', 'Instant Noodles Shrimp Creamy Tom Yum Flavour Jumbo Pack', 'Pack', 'Thailand', '5', '2013 #10', '2013', '#10'),
 (1588, 992, 'Prima Taste', 'Singapore Laksa La Mian', 'Pack', 'Singapore', '5', '2013 #1', '2013', '#1'),
 (2295, 285, 'Doll', 'Artificial Chicken', 'Pack', 'Hong Kong', '4.5', '2012 #9', '2012', '#9'),
 (2002, 578, 'Nongshim', 'Shin Ramyun Black', 'Pack', 'South Korea', '4.75', '2012 #7', '2012', '#7'),
 (2189, 391, 'Myojo', 'Hyoubanya No Chukasoba Oriental', 'Pack', 'Japan', '4.25', '2012 #6', '2012', '#6'),
 (2146, 434, 'Mi Sedaap', 'Kari Spesial', 'Pack', 'Indonesia', '4.5', '2012 #5', '2012', '#5'),
 (2567, 13, 'Sapporo Ichiban', 'Chow Mein', 'Pack', 'Japan', '5', '2012 #4', '2012', '#4'),
 (2188, 392, 'Nissin', 'Yakisoba Noodles Karashi', 'Tray', 'Japan', '5', '2012 #3', '2012', '#3'),
 (2533, 47, 'Indomie', 'Mi Goreng Jumbo Barbecue Chicken', 'Pack', 'Indonesia', '5', '2012 #2', '2012', '#2'),
 (1972, 608, 'Koka', 'Spicy Black Pepper', 'Pack', 'Singapore', '5', '2012 #10', '2012', '#10'),
 (2475, 105, 'Indom


It appears that the dataset is incomplete regarding the Top Ten rankings for each year.

This is not good as it may potentially affect the reliability of any insights derived from the data, leading to biased interpretations or incomplete understanding of trends or patterns

#### Question 9: I live in Singapore and I'm curious to know the kind of noodles produced here. So, what type of noodles are produced in Singapore?

In [20]:
%%sql

SELECT *
FROM ramen
WHERE Country == "Singapore"


 * sqlite:///ramen.db
Done.


[(9, 2571, 'KOKA', 'The Original Spicy Stir-Fried Noodles', 'Pack', 'Singapore', '2.5', None),
 (15, 2565, 'KOKA', 'Mushroom Flavour Instant Noodles', 'Cup', 'Singapore', '3.5', None),
 (22, 2558, 'KOKA', 'Creamy Soup With Crushed Noodles Hot & Sour Fish Flavor', 'Cup', 'Singapore', '5', None),
 (29, 2551, 'KOKA', 'Signature Tom Yum Flavor Instant Noodles', 'Pack', 'Singapore', '4', None),
 (35, 2545, 'KOKA', 'Instant Noodles Laksa Singapura Flavour', 'Pack', 'Singapore', '5', None),
 (37, 2543, 'KOKA', 'Curry Flavour Instant Noodles', 'Cup', 'Singapore', '5', None),
 (43, 2537, 'Nissin', 'Cup NoodlesTom Yam Seafood (More Shrimptacular)', 'Cup', 'Singapore', '5', None),
 (52, 2528, 'Prima', "Juzz's Mee Creamy Chicken Flavour (Export Version)", 'Pack', 'Singapore', '5', None),
 (56, 2524, 'Nissin', 'Straits Reborn Laksa', 'Pack', 'Singapore', '5', None),
 (59, 2521, 'Nissin', 'Cup Noodles Potato Chips Black Pepper Crab Flavour', 'Pack', 'Singapore', '4.5', None),
 (60, 2520, 'Nissin', 'Cup Noodles Blacl Pepper Crab Flavour (More Angry Crab)', 'Cup', 'Singapore', '5', None),
 (74, 2506, 'KOKA', 'Delight Chicken Flavor Instant Non-Fried Noodles', 'Pack', 'Singapore', '3.75', None),
 (81, 2499, 'KOKA', 'Creamy Soup With Crushed Noodles Chicken Flavor', 'Cup', 'Singapore', '5', None),
 (111, 2469, 'KOKA', 'Signature Curry Flavor Instant Noodles', 'Cup', 'Singapore', '3.5', None),
 (123, 2457, 'KOKA', 'Instant Noodles Chicken Satay Flavour', 'Pack', 'Singapore', '3.5', None),
 (124, 2456, 'KOKA', 'Signature Spicy Singapore Fried Noodles Instant Noodles', 'Pack', 'Singapore', '4.25', None),
 (125, 2455, 'KOKA', 'Creamy Soup With Crushed Noodles Sweet Corn Flavor', 'Cup', 'Singapore', '5', None),
 (126, 2454, 'KOKA', 'Silk Beef Pho Flavor Instant Rice Fettuccine', 'Bowl', 'Singapore', '3.25', None),
 (127, 2453, 'KOKA', 'Delight Spicy Sesame Flavor Instant Non-Fried Noodles', 'Pack', 'Singapore', '4.5', None),
 (128, 2452, 'KOKA', 'Signature Laksa Singapura Instant Noodles', 'Bowl', 'Singapore', '5', None),
 (129, 2451, 'KOKA', 'Delight Spicy Black Pepper Flavor Instant Non-Fried Noodles', 'Pack', 'Singapore', '5', None),
 (130, 2450, 'KOKA', 'Purple Wheat Noodles Chili & Lime Flavor', 'Pack', 'Singapore', '4', None),
 (131, 2449, 'KOKA', 'Delight Tomato Flavor Instant Non-Fried Noodles', 'Pack', 'Singapore', '5', None),
 (132, 2448, 'KOKA', 'Seafood Flavor Instant Noodles', 'Cup', 'Singapore', '3.5', None),
 (133, 2447, 'KOKA', 'Signature Stir-Fry Original Flavor Instant Noodles', 'Pack', 'Singapore', '3', None),
 (134, 2446, 'KOKA', 'Creamy Soup Witrh Crushed Noodles Curry Flavor', 'Cup', 'Singapore', '5', None),
 (135, 2445, 'KOKA', 'Delight Laksa Singapura Flavor Instant Non-Fried Noodles', 'Pack', 'Singapore', '4.25', None),
 (136, 2444, 'KOKA', 'Silk Spicy Marinara Instant Rice Fettuccine', 'Bowl', 'Singapore', '5', None),
 (137, 2443, 'KOKA', 'Signature Chicken Flavor Instant Noodles', 'Cup', 'Singapore', '3.75', None),
 (176, 2404, 'MyKuali', 'Penang Hokkien Prawn Noodle Soup', 'Pack', 'Singapore', '5', None),
 (180, 2400, 'Nissin', 'Cup Noodles Tom Yam Seafood Flavour', 'Cup', 'Singapore', '5', None),
 (357, 2223, 'KOKA', 'Rice Noodles Fragrant Chicken Pho Ga Flavour', 'Pack', 'Singapore', '4', None),
 (365, 2215, 'Myojo', 'Mee Goreng Spicy Flavour', 'Pack', 'Singapore', '5', None),
 (411, 2169, 'Happy Family', 'Vegetable Flavour Instant Noodles', 'Pack', 'Singapore', '4', None),
 (449, 2131, 'Myojo', 'Mee Goreng Thai Tom Yum Flavour', 'Pack', 'Singapore', '4.0', None),
 (455, 2125, 'Happy Family', 'Curry Flavour Instant Noodles', 'Pack', 'Singapore', '4', None),
 (480, 2100, 'Myojo', 'Mee Goreng Original Flavour', 'Pack', 'Singapore', '4', None),
 (485, 2095, 'Nissin', 'Straits Kitchen Laksa', 'Pack', 'Singapore', '5', None),
 (595, 1985, 'Prima Taste', 'Singapore Prawn Soup La Mian', 'Pack', 'Singapore', '5', None),
 (627, 1953, 'Prima Taste', 'Singapore Fish Soup La Mian', 'Pack', 'Singapore', '5', None),
 (633, 1947, 'Pri

#### Question 10: How many of Singapore's Ramen varities have made to the Top 10 Rankings??

In [21]:
%%sql

SELECT *
FROM ramen
WHERE Top_Ten IS NOT NULL
    AND Country == "Singapore";

 * sqlite:///ramen.db
Done.


[(633, 1947, 'Prima Taste', 'Singapore Laksa Wholegrain La Mian', 'Pack', 'Singapore', '5', '2016 #1'),
 (655, 1925, 'Prima', "Juzz's Mee Creamy Chicken Flavour", 'Pack', 'Singapore', '5', '2016 #8'),
 (673, 1907, 'Prima Taste', 'Singapore Curry Wholegrain La Mian', 'Pack', 'Singapore', '5', '2016 #5'),
 (1309, 1271, 'Prima Taste', 'Singapore Chilli Crab La Mian', 'Pack', 'Singapore', '5', '2014 #8'),
 (1588, 992, 'Prima Taste', 'Singapore Laksa La Mian', 'Pack', 'Singapore', '5', '2013 #1'),
 (1589, 991, 'Prima Taste', 'Singapore Curry La Mian', 'Pack', 'Singapore', '5', '2013 #2'),
 (1972, 608, 'Koka', 'Spicy Black Pepper', 'Pack', 'Singapore', '5', '2012 #10')]

Singapore boasts seven distinct varieties in the Top 10 rankings! Prima Taste has a total of five flavors, closely followed by Prima and Koka, which are tied.

## Conclusion
The ramen dataset on Kaggle provides comprehensive information about various ramen products, including their brand, variety, style, country of origin, and ratings. With over 2,500 entries, the dataset offers a diverse range of ramen products from different countries and brands. 

### Thank you for taking time to view my project! 😊